event_reporter/reporter/application_event_reporter.go

```diff
@@ -425,6 +403,59 @@ func (s *applicationEventReporter) processResource(
 	return nil
 }
 
+func (s *applicationEventReporter) getResourceActualState(ctx context.Context, logCtx *log.Entry, metricsEventType metrics.MetricEventType, rs appv1.ResourceStatus, parentApplication *appv1.Application, childApplication *appv1.Application) (*application.ApplicationResourceResponse, error) {
+	if isApp(rs) {
+		if childApplication.TypeMeta.Size() == 0 || childApplication.TypeMeta.Kind == "" || childApplication.TypeMeta.APIVersion == "" {
```

Comment:
lets move it to application struct as dedicated function. I know that it is argocd code, but it will looks better with such change